### ISSUE TITLE
Online Co-op: Room-loading sometimes failing, and sometimes unable to use waypoints

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#737]
New report from 0.2.1.

Room loading still not working properly at times, and leaving and rejoining makes waypoints unable to be used.

> The room loaded with only a black screen, I left and rejoined the lobby, and when we tried to leave it wouldn't allow us. We tried to use the warp station to go back to the main hub, but it kept giving us the message "Waiting for server..." We had to force quite the run because it trapped us in that room. Leaving and rejoining did nothing to stop the issue.

Feedback ID: 20201213-9YR48

Log from client:

```
7:41:11, Frame 30083, WARNING »  NETWORK CLIENT: RelayClient disconnected: RemoteConnectionClose
7:41:17, Frame 30441, LOG »  RELAY CLIENT: Connecting to server...
7:41:18, Frame 30487, LOG »  NETWORK CLIENT: Connecting to 207.246.89.219:5889 ...
7:41:18, Frame 30490, LOG »  NETWORK CLIENT: RelayClient connected to server 207.246.89.219:5889
7:41:19, Frame 30551, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPlayerHasFadedToBlack: Sending message to server...
7:41:19, Frame 30555, LOG »  NetworkMessage_DungeonRoomTransitionHelper: Received message. peekDesire: 4
7:41:20, Frame 30577, LOG »  DungeonRoomTransitioner.Update: Revealing room: Room 43: Fire Grotto
7:43:09, Frame 37140, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPartyShouldStartUsingWaypoint: Sending message to server...
[[[ ^ REPEATING MESSAGE... ]]]

7:43:39, Frame 38931, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPartyShouldStartUsingCorridor: Sending message to server...
[[[ ^ REPEATING MESSAGE... ]]]
```

Need to further improve robustness around:
- Ensuring players are loading into the correct rooms.
- Ensuring player netIDs and their creature netIDs have their parent room netIDs set correctly.

**Comment by ChrisNZL** [comment]
Additional report, v0.2.1, feedback ID: 20201213-Y8TQV

```
10:43:09, Frame 3913, LOG »  DungeonRoomTransitioner.Update: Revealing room: Room 43: Fire Grotto
10:43:29, Frame 5086, LOG »  RELAY CLIENT: Connecting to server...
10:43:29, Frame 5132, LOG »  NETWORK CLIENT: Connecting to 207.246.89.219:5889 ...
10:43:29, Frame 5135, LOG »  NETWORK CLIENT: RelayClient connected to server 207.246.89.219:5889
10:43:31, Frame 5211, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPlayerHasFadedToBlack: Sending message to server...
10:43:31, Frame 5216, LOG »  NetworkMessage_DungeonRoomTransitionHelper: Received message. peekDesire: 4
10:43:31, Frame 5238, LOG »  DungeonRoomTransitioner.Update: Revealing room: Room 43: Fire Grotto
10:44:14, Frame 7754, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPartyShouldStartUsingWaypoint: Sending message to server...
[[[ ^ REPEATING MESSAGE... ]]]
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.2.2.

Pre-0.2.2:
- Player's individual creatures could be assigned an object to spawn onto upon room load, and/or setting custom world spawn positions.
- Issues were detected where certain weapons and other events were setting said spawn points and positions individually... which was causing timing issues when rooms were loading, especially online.

Now in 0.2.2:
- Spawn points for players are now set on each dungeon room, rather than each player's creature, which makes the "where should the party spawn when the room loads?" system more robust.
- When the game sets which dungeon room is now the current room, the spawn point and position info is set immediately, and is longer set through other events.
